File: 145_postorder_BST/recur_simple.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> node_list;
    void postorder_trav(TreeNode* curr)
    {
        if (curr == NULL)
            return;
        
        postorder_trav(curr->left);
        postorder_trav(curr->right);
        node_list.push_back(curr->val);

    }
    vector<int> postorderTraversal(TreeNode* root) {
        if (root == NULL)
            return node_list;
            
        postorder_trav(root);
        return node_list;
    }
};

File: 145_postorder_BST/recur_simple.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    node_list = []
    def postorder_trav(self, curr):
        if curr == None:
            return

        self.postorder_trav(curr.left)
        self.postorder_trav(curr.right)
        self.node_list.append(curr.val)

    def postorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        self.node_list.clear()
        self.postorder_trav(root)
        return self.node_list